### ISSUE TITLE
Corrupted videos created by Intel Arc Control Center capturing on web

### ISSUE THREAD

**Issue by akostadinov** [immich-app/immich#14657]
### The bug

Please upload this image via browser https://filetransfer.io/data-package/mARMZgRx#link
Also allow HEVC 720p images not to need reencoding.

After uploading the image does not play. Shows an exclamation mark. Also when downloaded has a different sha512sum.

In the link above is the file with its expected SHA512.

### The OS that Immich Server is running on

Fedora 40

### Version of Immich Server

v1.121.0

### Version of Immich Mobile App

NA

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
I have converted the compose items into systemd services. Using quadlet for a systemd user services with rootless podman. But basically it is the v1.121.0 version of the compose converted with some a little change in how postgres env variables are handled (but the database works fine, it is accessible and shows healthy check status).


;/etc/containers/systemd/users/1002/immich-redis.container
[Container]
ContainerName=immich_redis
Network=immich
NetworkAlias=redis
HealthCmd=redis-cli ping || exit 1
Image=docker.io/redis:6.2-alpine@sha256:e3b17ba9479deec4b7d1eeec1548a253acc5374d68d3b27937fcfe4df8d18c7e

[Service]
TimeoutStartSec=900
Restart=always

;/etc/containers/systemd/users/1002/immich-pgsql.container
[Container]
ContainerName=immich_postgres
Environment=POSTGRES_INITDB_ARGS=--data-checksums
EnvironmentFile=/media/toshsec/containers_services_data/immich/pgsql.env
Exec=postgres -c 'shared_preload_libraries=vectors.so' -c 'search_path="$user", public, vectors' -c 'logging_collector=off' -c 'log_destination=stderr' -c 'max_wal_size=2GB' -c 'shared_buffers=512MB' -c 'wal_compression=on'
HealthCmd=pg_isready --dbname=$${POSTGRES_DB} --username=$${POSTGRES_USER} || exit 1; Chksum="$$(psql --dbname=$${POSTGRES_DB} --username=$${POSTGRES_USER} --tuples-only --no-align --command='SELECT COALESCE(SUM(checksum_failures), 0) FROM pg_stat_database')"; echo "checksum failure count is $$Chksum"; [ "$$Chksum" = '0' ] || exit 1
HealthInterval=5m
HealthStartPeriod=5m
HealthStartupInterval=30s
Network=immich
NetworkAlias=postgres
Image=docker.io/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
Volume=/media/toshsec/containers_services_data/immich/postgresql/data:/var/lib/postgresql/data,Z

[Service]
TimeoutStartSec=900
Restart=always

/etc/containers/systemd/users/1002/immich-machine-learning.container 
[Container]
ContainerName=immich_machine_learning
# EnvironmentFile=/media/toshsec/containers_services_data/immich/immich.env
Image=ghcr.io/immich-app/immich-machine-learning:v1.121.0
Network=immich,podman
DNS=none
NetworkAlias=immich-machine-learning
Volume=immich-model-cache.volume:/cache

[Service]
TimeoutStartSec=900
Restart=always

;/etc/containers/systemd/users/1002/immich-server.container
[Unit]
Requires=immich-redis.service immich-pgsql.service
After=immich-redis.service immich-pgsql.service
Wants=immich-machine-learning

[Container]
ContainerName=immich_server
EnvironmentFile=/media/toshsec/containers_services_data/immich/immich.env
Image=ghcr.io/immich-app/immich-server:v1.121.0
PublishPort=8102:2283
Network=immich
StopTimeout=900
GroupAdd=keep-groups
AddDevice=/dev/dri:/dev/dri
Volume=/media/toshsec/containers_services_data/immich/library:/usr/src/app/upload:Z
Volume=/etc/localtime:/etc/localtime:ro

[Install]
WantedBy=multi-user.target default.target

[Service]
TimeoutStartSec=900
TimeoutStopSec=950
Restart=always
```

### Your .env content

```Shell
#immich.env
DB_PASSWORD=REDACTED
DB_HOSTNAME=postgres
DB_USERNAME=postgres
DB_DATABASE_NAME=immich

#pgsql.env
POSTGRES_PASSWORD=REDACTED
POSTGRES_USER=postgres
POSTGRES_DB=immich
```

### Reproduction steps

1. upload video from download link
2. try to open video
3. download video and verify SHA512

I'm not sure it always happens but does for me. So in case it doesn't happen for you with one upload, please try a couple more times. I noticed the problem uploading other smaller videos captured by the Intel Arc Control _fine_ software. And I noticed that smaller ones succeeded easily. This is a test recording of something irrelevant just to be able to share it in public.

### Relevant log output

```shell
I see nothing interesting in the log. Just these two lines when I upload the file:

[Nest] 2  - 12/11/2024, 9:42:14 PM    WARN [Microservices:MetadataService] Asset 903662c5-b6a0-4b9e-adc8-783dc31238a8 has no valid date, falling back to asset.fileCreatedAt
[Nest] 2  - 12/11/2024, 9:42:15 PM   ERROR [Microservices:MediaService] mov,mp4,m4a,3gp,3g2,mj2
```
```


### Additional information

I'm sure Intel Arc Control is doing something wrong, still immich shouldn't corrupt or alter the uploaded files as far as I understand (and expect).

**Comment by bo0tzz** [comment]
I can't reproduce this. I do get the playback error - which is somewhat expected if you're disabling transcoding - but when I hit the Download original file button I get the exact same file back:
```
f79057dd21c637d6e8caa1fd140aeb60d3b4490e2085b9efa09553115df6d797d9b5dec3d8ab01f4999bad439799f09687026ebfad2d7ff11297607b4d81eee0  test-arc-control.mp4
f79057dd21c637d6e8caa1fd140aeb60d3b4490e2085b9efa09553115df6d797d9b5dec3d8ab01f4999bad439799f09687026ebfad2d7ff11297607b4d81eee0  test-arc-control(1).mp4
```

**Comment by akostadinov** [comment]
> I do get the playback error - which is somewhat expected if you're disabling transcoding

Why is it expected? HEVC mp4 file with HD resolution? Shouldn't that work without transcoding?

Now I performed further testing. With Chrome the audio play but not the video. Is there any information what the file format should match so it is playable across all platforms - web, android and iOS? 

> I can't reproduce this.

I did some more testing. It turns out that the issue is during download. On the server I see the file with the correct checksum.

 With Firefox the download is clipped to ~ 2GB. It is also strange that both beginning and end of files don't match:
```
$ diff <(head -c 200000 test-arc-control.mp4) <(head -c 20000 test-arc-control-downloaded.mp4)
Binary files /dev/fd/63 and /dev/fd/62 differ
$ diff <(tail -c 200000 test-arc-control.mp4) <(tail -c 20000 test-arc-control-downloaded.mp4)
Binary files /dev/fd/63 and /dev/fd/62 differ
```
So not sure how to understand this.

On Chrome I can't download at all but only see
![image](https://github.com/user-attachments/assets/e181ae77-0477-4f69-8c9e-ea66e5576122)

Is there a special way to download the original file? I though download is always downloading the original. Or am I mistaken?

**Comment by bo0tzz** [comment]
> Why is it expected?

If you tell Immich to not transcode a file, it just serves whatever was uploaded and doesn't have anything to do with whether the client will be able to play it.

> With Firefox the download is clipped to ~ 2GB
> On Chrome I can't download at all 

This sounds like a reverse proxy issue to me. Try accessing directly on ip:port.

**Comment by akostadinov** [comment]
In fact - yes! Accessing the port directly did allow proper download with both - Firefox and Chrome.

It is strange though that with the reverse proxy, download completed without issues, and file was smaller and corrupt. Because the correct size is visible in the download popup, there are no errors shown, just file written is different.

I think this is worse than on Chrome, where it shows a failure outright. Can such condition be detected by the immich web app so that one knows a downloaded file is correct? Lets say I fix or change the reverse proxy now. I test that the file works alright. But then how can I detect if new proxy starts to misbehave again?

Can the download functionality be smarter in detecting this?

For a reference I download large files from seafile on the same server through the same reverse proxy (rust-rpxy) and I don't observe corruption. But when I download a large file from seafile, the download happens through the standard browser download functionality. With immich, I see this separate popup, once it is finished, then standard browser download starts and completes very quickly. So apparently immich downloads in some more tricky way.

Some insight why immich functions like that and some more information how to debug this would be very appreciated. Because now I have no idea what might the proxy be doing wrong (or unexpected for immich).

**Comment by akostadinov** [comment]
@bo0tzz , can this issue be reopened now I understood what the issue is and changed description or should I file a new issue?

**Comment by bo0tzz** [comment]
I'm not convinced that this is an Immich issue. It seems more like an incognito mode limitation to me - I can't find any clear documentation, but I believe that comes with a bunch of resource limits that could very well include download size. It's also conceivable that it's related to #14725, though I'm not sure about that.

**Comment by akostadinov** [comment]
Ok, thank you, I think #14725 should address this. There is not issue downloading large files in incognito mode but this saving to memory first. I'm surprised downloading multi-GB files works at all in normal mode. Thank you for pointing this out!
